zcash_primitives/src/consensus.rs

```diff
@@ -301,6 +301,14 @@ impl Parameters for TestNetwork {
 pub enum Network {
     MainNetwork,
     TestNetwork,
+    RegtestNetwork {
+        overwinter: Option<BlockHeight>,
+        sapling: Option<BlockHeight>,
+        blossom: Option<BlockHeight>,
+        heartwood: Option<BlockHeight>,
+        canopy: Option<BlockHeight>,
+        nu5: Option<BlockHeight>,
+    },
```

Comment:
Thanks @str4d for your review. I agree with this quote below.
> I think regtest support (or more generally, user-configurable test networks) should be provided with a separate type that downstreams can opt into, similar to what we do in zcashd, and that lives somewhere other than zcash_primitives::consensus (maybe zcash_primitives::testing)

The only thing I would differ is that we need to be able to use regtest from clients to have the integration tests like the advanced ReOrg tests. So we would need to access this RegtestNetwork types from the FFI.

I chose consensus::Network mostly for two reasons:
1. other crates were referring to it 
2. [The FFI uses that](https://github.com/Electric-Coin-Company/zcash-light-client-ffi/blob/310c1ec43d59fa7b6a2962e30fd344c028bd1291/rust/src/lib.rs#L54-L68)

And no others, so I'm definitely ok with refactoring this so that it's more convenient for today's reality.

Reply:
> The FFI uses it 

To clarify, the FFI itself uses numeric constants for the kinds of networks it knows about, and then [converts those into a `consensus::Network`](https://github.com/Electric-Coin-Company/zcash-light-client-ffi/blob/310c1ec43d59fa7b6a2962e30fd344c028bd1291/rust/src/lib.rs#L2686). So the way for the iOS SDK to opt in to testing with a configurable local network is to change that helper function to use the new type that supports local testing. That change makes sense because the iOS SDK won't be able to do local testing without a bunch of other changes on its Swift side.

> other crates were referring to it

If we change `consensus::Network` then some other crates using it may be forced to do that same bunch of work (depending on how they use the enum) that they don't want to do (perhaps because they are only concerned with the consensus networks).

Reply:
@str4d I think the changes I pushed might be heading into the right direction